dts: msm8953: Add Lenovo ThinkSmart View (CD-18781Y)

kaechele opened the pull request:
Menu navigation is currently not possible as the only two buttons available on this device are Volume Up and Volume Down. Menu items can be browsed but not selected as there is no button mapped to KEY_HOME.

Note that bootloader unlock is not functional on the ThinkSmart View. But the device will boot any image as long as it is AVB1 signed. For simple and quick signing `magiskboot sign` can be used on `lk2nd.img`.

TravMurav commented on lk2nd/device/dts/msm8953/apq8053-lenovo-cd-18781y.dts:
```diff
@@ -0,0 +1,31 @@
+	model = "Lenovo ThinkSmart View";
+	compatible = "lenovo,cd-18781y";
+
+    lk2nd,dtb-files = "apq8053-lenovo-cd-18781y";
```
This is spaces not tabs, I'd appreciate if everything is consistent and uses tabs.

TravMurav commented on lk2nd/device/dts/msm8953/apq8053-lenovo-cd-18781y.dts:
```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0-only
```
I'd appreciate new submissions being `BSD-3-Clause` like what upstream Linux prefers

TravMurav commented on lk2nd/device/dts/msm8953/apq8053-lenovo-cd-18781y.dts:
```diff
@@ -0,0 +1,31 @@
+	compatible = "lenovo,cd-18781y";
+
+    lk2nd,dtb-files = "apq8053-lenovo-cd-18781y";
+
```
> the only two buttons available on this device are Volume Up and Volume Down.

Then you may want to map vol+ to KEY_POWER keycode using gpio-keys, the menu loops around so it's fine but the message on the screen will be slightly incorrect. Better than having no menu though

kaechele replied:
I've tried it and it doesn't actually work. Even with the `gpio-keys` configuration in there the Volume Up key still simply navigates through the menu.

TravMurav replied:
Ah, sorry, it's a subtle bug/quirk in the gpio-keys/menu implementation: it checks for the volume keys first and since there is no node for vol+, it uses the default key, which then suppresses the power keycode... 

As a hack, I think you can add a vol+ key with `gpios = <&pmic_pon 0 0>;` to suppress the keycode with an non-existing key, please add a HACK: comment near it if you do

kaechele replied:
That worked, thanks!

TravMurav reviewed (commented):
That's a quite funny device, please also edit the devices.md list in the same commit to add it, Probably want to add some extra notes that the device is "quirky" and document how to fixup the lk2nd.img to let it boot it, may also add a note that the vol+ is mapped to power key for lk2nd. See how other devices document this.

kaechele commented:
Thanks for the review. I've addressed most comments. Key remapping doesn't seem to work. I've committed the configuration I used in the dts file for review purposes, but it doesn't actually function as intended.

TravMurav commented:
Thanks!
